chain-signatures/node/src/protocol/presignature.rs

```diff
@@ -23,12 +27,94 @@ use near_account_id::AccountId;
 pub type PresignatureId = u64;
 
 /// A completed presignature.
+#[derive(Clone)]
 pub struct Presignature {
     pub id: PresignatureId,
     pub output: PresignOutput<Secp256k1>,
     pub participants: Vec<Participant>,
 }
 
+impl Default for Presignature {
+    fn default() -> Self {
+        Self {
+            id: 1,
+            output: PresignOutput {
+                big_r: <Secp256k1 as CurveArithmetic>::AffinePoint::default(),
+                k: <Secp256k1 as CurveArithmetic>::Scalar::ZERO,
+                sigma: <Secp256k1 as CurveArithmetic>::Scalar::ONE,
+            },
+            participants: vec![Participant::from(1), Participant::from(2)],
+        }
+    }
+}
+
+impl Serialize for Presignature {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut state = serializer.serialize_struct("Presignature", 3)?;
+        state.serialize_field("id", &self.id)?;
+
+        let mut output_map = serde_json::Map::new();
+        output_map.insert(
+            "big_r".to_string(),
+            serde_json::to_value(self.output.big_r).map_err(serde::ser::Error::custom)?,
+        );
+        output_map.insert(
+            "k".to_string(),
+            serde_json::to_value(self.output.k).map_err(serde::ser::Error::custom)?,
+        );
+        output_map.insert(
+            "sigma".to_string(),
+            serde_json::to_value(self.output.sigma).map_err(serde::ser::Error::custom)?,
+        );
```

Comment:
we shouldn't be encoding this with serde_json as that has overhead. Is there a specific reason why we don't just serialize each field of output?
```
        let mut state = serializer.serialize_struct("Presignature", 5)?;
        state.serialize_field("id", &self.id)?;
        state.serialize_field("output_big_r", &self.output.big_r)?;
        state.serialize_field("output_k", &self.output.k)?;
        state.serialize_field("output_sigma", &self.output.sigma)?;
        state.serialize_field("participants", &self.participants)?;
        state.end()
```

Reply:
agree, simlified